python/mlc_chat/cli/compile.py

```diff
@@ -139,6 +139,22 @@ def _check_prefix_symbols(prefix: str) -> str:
         "4) Android, iOS: .tar (objects); "
         "5) Web: .wasm (web assembly)",
     )
+    parser.add_argument(
```

Comment:
Just curious (not proposing changes) - do those two parameters affect compile-time or runtime behavior?

I did notice that the default initial length of KVCache is controlled by `--sliding-window`, and originally in Llama by `--max-sequence-length`, both of which I believe could be instead moved to runtime behavior to avoid potential re-compilation upon hyperparameter change.

Reply:
I agree it is a bit cumbersome to recompile for different hyperparameters. 

As of now, `sliding_window` affects both compile-time and runtime, while the chunk size only affects runtime. The sliding window affects the [creation of an attention mask](https://github.com/mlc-ai/mlc-llm/pull/1230/files#diff-b1e30e2e75906e7cebc86215ebcc688e8d102ad9552d585dc9537223a4527e7dR223), and some other things like calculating where in the cache to override.

Another compile-time thing that all three variables would affect is the [variable bound](https://github.com/mlc-ai/mlc-llm/blob/main/python/mlc_chat/compiler/compile.py#L85).

Reply:
Another question: consider the case that we expand chunking to all LLMs, not limiting to Mistral, do you think it can be used to decide the upper bound of `seq_len`?

Reply:
Yep! With some sacrifice of prefill speed (no effect on generated output at all), `seq_len` should be able to be bounded precisely by `chunk_size` like here: 
https://github.com/mlc-ai/mlc-llm/blob/main/mlc_llm/relax_model/mistral.py#L976-L982.

Reply:
Would be very interesting to learn what a reasonable `chunk_size` should be without sacrificing performance. I imagine something like 512 or 1024 on CUDA GPU should be good enough